src/lib/dlt_user.c

```diff
@@ -1002,46 +1002,38 @@ DltReturnValue dlt_register_app(const char *apid, const char *description)
         return DLT_RETURN_OK;
     }
 
-    dlt_user.reference++;
+    dlt_user.app_reference++;
```

Comment:
I think you should set it to 1 instead of increase (if you still want to increase, you have to set it to 0 instead of decrease in case of dlt_unregister_app) and only after dlt_user_log_send_register_application() return OK at line 1033 to make sure the application is registered successfully.